samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -72,7 +96,10 @@ public ContainerManager(SamzaApplicationState samzaApplicationState, ClusterReso
    */
   void handleContainerLaunch(SamzaResourceRequest request, String preferredHost, SamzaResource allocatedResource,
       ResourceRequestState resourceRequestState, ContainerAllocator allocator) {
-    if (this.standbyContainerManager.isPresent()) {
+    if (getControlActionMetadata(request.getProcessorId()).isPresent()) {
```

Comment:
expose a suitable method e.g., isControlActionRequest(request) (perhaps a wrapper on getControlActionMetadata), 
It'd make the logic more understanbable and later if you need to add additional validations (e.g., timestamp based) one can add it there.